File: backend/src/modules/domain/domain.controller.ts
import {
  Controller,
  Get,
  Post,
  Body,
  Patch,
  Param,
  Delete,
  Query,
  ParseIntPipe,
} from '@nestjs/common';
import { ApiTags, ApiOperation, ApiResponse } from '@nestjs/swagger';
import { DomainService } from './domain.service';
import { CreateDomainDto } from './dto/create-domain.dto';
import { UpdateDomainDto } from './dto/update-domain.dto';
import { PaginationDto } from '../../common/dto/pagination.dto';
import { ApiResponseDto } from '../../common/dto/api-response.dto';
import { Domain } from './entities/domain.entity';

@ApiTags('域管理')
@Controller('api/domains')
export class DomainController {
  constructor(private readonly domainService: DomainService) {}

  @Post()
  @ApiOperation({ summary: '创建域' })
  @ApiResponse({ status: 201, description: '创建成功' })
  async create(@Body() createDto: CreateDomainDto): Promise<ApiResponseDto<Domain>> {
    const domain = await this.domainService.create(createDto);
    return ApiResponseDto.success(domain, '创建成功');
  }

  @Get()
  @ApiOperation({ summary: '获取域列表' })
  @ApiResponse({ status: 200, description: '获取成功' })
  async findAll(@Query() paginationDto: PaginationDto): Promise<ApiResponseDto> {
    const result = await this.domainService.findAll(paginationDto);
    return ApiResponseDto.success(result);
  }

  @Get('all')
  @ApiOperation({ summary: '获取所有域（不分页）' })
  @ApiResponse({ status: 200, description: '获取成功' })
  async findAllDomains(): Promise<ApiResponseDto<Domain[]>> {
    const domains = await this.domainService.findAllDomains();
    return ApiResponseDto.success(domains);
  }

  @Get(':id')
  @ApiOperation({ summary: '获取域详情' })
  @ApiResponse({ status: 200, description: '获取成功' })
  async findOne(@Param('id', ParseIntPipe) id: number): Promise<ApiResponseDto<Domain>> {
    const domain = await this.domainService.findOne(id);
    return ApiResponseDto.success(domain);
  }

  @Patch(':id')
  @ApiOperation({ summary: '更新域' })
  @ApiResponse({ status: 200, description: '更新成功' })
  async update(
    @Param('id', ParseIntPipe) id: number,
    @Body() updateDto: UpdateDomainDto,
  ): Promise<ApiResponseDto<Domain>> {
    const domain = await this.domainService.update(id, updateDto);
    return ApiResponseDto.success(domain, '更新成功');
  }

  @Delete(':id')
  @ApiOperation({ summary: '删除域' })
  @ApiResponse({ status: 200, description: '删除成功' })
  async remove(@Param('id', ParseIntPipe) id: number): Promise<ApiResponseDto> {
    await this.domainService.remove(id);
    return ApiResponseDto.success(null, '删除成功');
  }

  @Post('reload')
  @ApiOperation({ summary: '重载域配置' })
  @ApiResponse({ status: 200, description: '重载成功' })
  async reload(): Promise<ApiResponseDto> {
    await this.domainService.reload();
    return ApiResponseDto.success(null, '重载成功');
  }
}

File: kam-admin-module-user/src/main/java/io/kamailio/admin/modules/domain/controller/DomainController.java
package io.kamailio.admin.modules.domain.controller;

import io.kamailio.admin.common.dto.ApiResponse;
import io.kamailio.admin.common.dto.PaginatedResult;
import io.kamailio.admin.common.dto.PaginationDto;
import io.kamailio.admin.modules.domain.dto.CreateDomainDto;
import io.kamailio.admin.modules.domain.dto.UpdateDomainDto;
import io.kamailio.admin.modules.domain.entity.Domain;
import io.kamailio.admin.modules.domain.service.DomainService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@Tag(name = "域管理")
@RestController
@RequestMapping("/api/domains")
public class DomainController {
    private final DomainService service;

    public DomainController(DomainService service) { this.service = service; }

    @PostMapping
    @Operation(summary = "创建域")
    public ResponseEntity<ApiResponse<Domain>> create(@Valid @RequestBody CreateDomainDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.create(dto), "创建成功"));
    }

    @GetMapping
    @Operation(summary = "获取域列表")
    public ResponseEntity<ApiResponse<PaginatedResult<Domain>>> findAll(PaginationDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.findAll(dto)));
    }

    @GetMapping("/all")
    @Operation(summary = "获取所有域（不分页）")
    public ResponseEntity<ApiResponse<List<Domain>>> findAllDomains() {
        return ResponseEntity.ok(ApiResponse.success(service.findAllDomains()));
    }

    @GetMapping("/{id}")
    @Operation(summary = "获取域详情")
    public ResponseEntity<ApiResponse<Domain>> findOne(@PathVariable Integer id) {
        return ResponseEntity.ok(ApiResponse.success(service.findOne(id)));
    }

    @PatchMapping("/{id}")
    @Operation(summary = "更新域")
    public ResponseEntity<ApiResponse<Domain>> update(@PathVariable Integer id, @Valid @RequestBody UpdateDomainDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.update(id, dto), "更新成功"));
    }

    @DeleteMapping("/{id}")
    @Operation(summary = "删除域")
    public ResponseEntity<ApiResponse<Void>> remove(@PathVariable Integer id) {
        service.remove(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    @PostMapping("/reload")
    @Operation(summary = "重载域配置")
    public ResponseEntity<ApiResponse<Void>> reload() {
        service.reload();
        return ResponseEntity.ok(ApiResponse.success(null, "重载成功"));
    }

    @GetMapping("/dump")
    @Operation(summary = "导出内存域列表")
    public ResponseEntity<ApiResponse<Object>> dump() {
        return ResponseEntity.ok(ApiResponse.success(service.dump()));
    }
}
